tests/postgres/mod.rs

```diff
@@ -0,0 +1,104 @@
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use futures::Future;
+use warg_client::{api, Config};
+use warg_server::datastore::{DataStore, PostgresDataStore};
+
+fn data_store() -> Result<Box<dyn DataStore>> {
+    Ok(Box::new(PostgresDataStore::new(
+        std::env::var("DATABASE_URL")
+            .context("failed to get `DATABASE_URL` environment variable")?,
+    )?))
+}
+
+async fn run<F>(root: &Path, callback: impl FnOnce(Config) -> F) -> Result<()>
+where
+    F: Future<Output = Result<()>>,
+{
+    let (_server, config) = super::support::spawn_server(root, Some(data_store()?)).await?;
+    callback(config).await
+}
+
+/// This test assumes the database is empty on each run.
+/// Use the `ci/run-postgres-tests.sh` script to run this test.
+///
+/// A smoke test that ensures that postgres integration works.
+///
+/// This is one large test currently because there can be only one core service
```

Comment:
Is there a reason we can't run two completely independent servers in parallel?

Reply:
Each server would independently have their own log and map cryptographic primitives to checkpoint from; thus the transparency services of each running server would have disjoint leafs (consequently, different log lengths and roots).

If you don't anticipate a problem with that, then I don't see a reason why not then.

If that would be a problem and we do require a singular transparency service, then I think we'll need to refactor the core service out to another process and use a proper (and reliable) message broker between them.

Reply:
We do need a single transparency service per registry instance, but I wasn't sure what this comment was getting at. It doesn't seem to me like there's anything keeping you from splitting this test into multiple tests that initialize their own completely independent registry instances and then each test one thing (like publish/validate). It does mean that you'd spin up several whole registries and their own databases, but if that's the problem I would write the comment differently.

Reply:
Oh gotcha! Yeah, the comment is in the context that the tests are assuming there's a single _database_ being communicated with for all of the tests, as the CI script just spawns a single container. Thus while we can start multiple servers to test against concurrently, the data would be shared across them with inconsistent core service state.

I can update the comment to make it less confusing.